src/components/Dropdown/DropdownItem.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+
+import { UIComponent, RenderResultConfig, createShorthandFactory, customPropTypes } from '../../lib'
+import {
+  Extendable,
+  ShorthandValue,
+  ShorthandRenderFunction,
+  ComponentEventHandler,
+} from '../../../types/utils'
+import { UIComponentProps } from '../../lib/commonPropInterfaces'
+import { commonUIComponentPropTypes } from '../../lib/commonPropTypes'
+import ListItem from '../List/ListItem'
+import Image from '../Image/Image'
+
+export interface DropdownItemProps extends UIComponentProps<any, any> {
+  /** Text content for the item description. */
+  content?: string
+
+  /** The main text displayed for the item. */
+  header?: string
+
+  /** The item is highlighted inside the list. */
+  highlighted?: boolean
+
+  /** The item can have an image avatar. */
+  image?: ShorthandValue
+
+  /**
+   * Called when clicking on the dropdown item in the list.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onClick?: ComponentEventHandler<DropdownItemProps>
+
+  /**
+   * A custom render function the image slot.
+   *
+   * @param {React.ReactType} Component - The computed component for this slot.
+   * @param {object} props - The computed props for this slot.
+   * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+   */
+  renderImage?: ShorthandRenderFunction
+}
+
+/**
+ * A DropdownItem is a sub-component if the Dropdown.
+ */
+class DropdownItem extends UIComponent<Extendable<DropdownItemProps>, any> {
```

Comment:
Add these sub-components as a static props in the Dropdown, so that they can be used in the following manner: `Dropdown.Item`, `Dropdown.Label` etc. Also, would encourage you to add some more descriptive message for what the component actually is, rather then it is a sub-component of the Dropdown.

Reply:
thank you, addressed it!